### PR TITLE
fix(api): unable to load workflow_node_run because of context canceled

### PR DESCRIPTION
**Pull request by fsamin**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/api/workflow/run_workflow_test.go [hunk 1]
```diff
@@ -367,7 +368,7 @@ func TestManualRun3(t *testing.T) {
 		}
 
 		//TakeNodeJobRun
-		j, _, err = workflow.TakeNodeJobRun(nil, db, db, cache, proj, j.ID, "model", "worker", "1", []sdk.SpawnInfo{
+		j, _, err = workflow.TakeNodeJobRun(nil, func() *gorp.DbMap { return db }, db, cache, proj, j.ID, "model", "worker", "1", []sdk.SpawnInfo{
```
Comment (golangci) [inline engine/api/workflow/run_workflow_test.go]: ineffectual assignment to `err`

File: engine/api/workflow/dao_run_test.go [hunk 2]
```diff
@@ -109,7 +109,7 @@ func TestPurgeWorkflowRun(t *testing.T) {
 	test.NoError(t, err)
 
 	for i := 0; i < 5; i++ {
-		_, _, errWr := workflow.ManualRun(nil, db, db, cache, proj, w1, &sdk.WorkflowNodeRunManual{
+		_, _, errWr := workflow.ManualRun(nil, db, cache, proj, w1, &sdk.WorkflowNodeRunManual{
```
Comment (golangci) [inline engine/api/workflow/dao_run_test.go]: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

File: engine/api/workflow/dao_run_test.go [hunk 3]
```diff
@@ -183,7 +183,7 @@ func TestPurgeWorkflowRunWithOneSuccessWorkflowRun(t *testing.T) {
 	})
 	test.NoError(t, err)
 
-	_, _, errWr := workflow.ManualRun(nil, db, db, cache, proj, w1, &sdk.WorkflowNodeRunManual{
+	_, _, errWr := workflow.ManualRun(nil, db, cache, proj, w1, &sdk.WorkflowNodeRunManual{
```
Comment (golangci) [inline engine/api/workflow/dao_run_test.go]: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

File: engine/api/workflow/dao_run_test.go [hunk 4]
```diff
@@ -193,7 +193,7 @@ func TestPurgeWorkflowRunWithOneSuccessWorkflowRun(t *testing.T) {
 	test.NoError(t, errWr)
 
 	for i := 0; i < 5; i++ {
-		wfr, _, errWr := workflow.ManualRun(nil, db, db, cache, proj, w1, &sdk.WorkflowNodeRunManual{
+		wfr, _, errWr := workflow.ManualRun(nil, db, cache, proj, w1, &sdk.WorkflowNodeRunManual{
```
Comment (golangci) [inline engine/api/workflow/dao_run_test.go]: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

File: engine/api/workflow/run_workflow_test.go [hunk 5]
```diff
@@ -124,7 +126,7 @@ func TestManualRun1(t *testing.T) {
 	test.Equal(t, 2, len(jobs))
 
 	//TestprocessWorkflowRun
-	wr2, _, err := workflow.ManualRunFromNode(nil, db, db, cache, proj, w1, 2, &sdk.WorkflowNodeRunManual{User: *u}, w1.RootID)
+	wr2, _, err := workflow.ManualRunFromNode(nil, db, cache, proj, w1, 2, &sdk.WorkflowNodeRunManual{User: *u}, w1.RootID)
```
Comment (golangci) [inline engine/api/workflow/run_workflow_test.go]: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

File: engine/api/workflow/run_workflow_test.go [hunk 6]
```diff
@@ -218,16 +220,16 @@ func TestManualRun2(t *testing.T) {
...
 	}, nil)
 	test.NoError(t, err)
 
-	_, _, err = workflow.ManualRun(nil, db, db, cache, proj, w1, &sdk.WorkflowNodeRunManual{User: *u}, nil)
+	_, _, err = workflow.ManualRun(context.TODO(), db, cache, proj, w1, &sdk.WorkflowNodeRunManual{User: *u}, nil)
 	test.NoError(t, err)
 
 	//TestprocessWorkflowRun
-	_, _, err = workflow.ManualRunFromNode(nil, db, db, cache, proj, w1, 1, &sdk.WorkflowNodeRunManual{User: *u}, w1.RootID)
+	_, _, err = workflow.ManualRunFromNode(nil, db, cache, proj, w1, 1, &sdk.WorkflowNodeRunManual{User: *u}, w1.RootID)
```
Comment (golangci) [inline engine/api/workflow/run_workflow_test.go]: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

File: engine/api/workflow/run_workflow_test.go [hunk 7]
```diff
@@ -367,7 +369,7 @@ func TestManualRun3(t *testing.T) {
 		}
 
 		//TakeNodeJobRun
-		j, _, err = workflow.TakeNodeJobRun(nil, db, db, cache, proj, j.ID, "model", "worker", "1", []sdk.SpawnInfo{
+		j, _, _ = workflow.TakeNodeJobRun(nil, func() *gorp.DbMap { return db }, db, cache, proj, j.ID, "model", "worker", "1", []sdk.SpawnInfo{
```
Comment (golangci) [inline engine/api/workflow/run_workflow_test.go]: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

File: engine/api/workflow/run_workflow_test.go [hunk 8]
```diff
@@ -410,7 +412,7 @@ func TestManualRun3(t *testing.T) {
 		}
 
 		//TestUpdateNodeJobRunStatus
-		_, err = workflow.UpdateNodeJobRunStatus(nil, db, db, cache, proj, j, sdk.StatusSuccess)
+		_, err = workflow.UpdateNodeJobRunStatus(nil, func() *gorp.DbMap { return db }, db, cache, proj, j, sdk.StatusSuccess)
```
Comment (golangci) [inline engine/api/workflow/run_workflow_test.go]: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

File: engine/api/workflow_run_test.go [hunk 9]
```diff
@@ -97,14 +98,14 @@ func Test_getWorkflowNodeRunHistoryHandler(t *testing.T) {
 	w1, err := workflow.Load(db, api.Cache, proj, "test_1", u, workflow.LoadOptions{})
 	test.NoError(t, err)
 
-	wr, _, errMR := workflow.ManualRun(nil, db, db, api.Cache, proj, w1, &sdk.WorkflowNodeRunManual{
+	wr, _, errMR := workflow.ManualRun(nil, db, api.Cache, proj, w1, &sdk.WorkflowNodeRunManual{
```
Comment (golangci) [inline engine/api/workflow_run_test.go]: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

**Comment by bnjjj** [comment]
CDS Report build-ui#4868.0

* Compile
  * Build UI ✔ 
  * TU model ✔ 
  * TU service ✔ 
  * TU shared ✔ 
  * TU views ✔ 

* Package
  * Package UI ✔ 

Unit Tests Report
* HeadlessChrome 0.0.0 (Linux 0.0.0)
  * CDS: Action Component should create and then delete a requirement ✘
  * CDS: Action Component should create and then delete a parameter ✘
  * CDS: Action Component should send delete action event ✘
  * CDS: Action Component should send insert action event ✘
  * CDS: Action Component should add and then remove a step ✘
  * CDS: Action Component should init step not always executed and step always executed ✘
  * CDS: Step Form Component should send add step event ✘
  * CDS: Action Component should send remove step event ✘
  * CDS: Delete Button Test emit delete event ✘
  * CDS: Commit List should load component ✘
  * CDS: Group form component should load component and disable button ✘
  * CDS: Group form component should load component and enable button ✘
  * CDS: parameter From Component should create a new parameter ✘
  * CDS: Parameter List Component should load component &#43; update description ✘
  * CDS: Parameter Value Component should create an input text ✘
  * CDS: Parameter Value Component should create an input number ✘
  * CDS: Parameter Value Component should create a checkbox ✘
  * CDS: Parameter Value Component should create a select for pipeline ✘
  * CDS: Parameter Value Component should create a select for environments ✘
  * CDS: Permission From Component should create new permission ✘
  * CDS: Permission List Component should delete a permission ✘
  * CDS: Permission List Component should update a permission ✘
  * CDS: Permission List Component should get permission name by value ✘
  * CDS: prerequisite From Component should create a new prerequisite ✘
  * CDS: Prerequisite List Component should load component &#43; update value ✘
  * CDS: Project RepoManager Form Component Add new repo manager ✘
  * CDS: Requirement Form Component should create a new requirement and auto write name ✘
  * CDS: Requirement List Component should load component &#43; delete requirement ✘
  * CDS: Parameter Value Component should display success icon ✘
  * CDS: Parameter Value Component should display building loader ✘
  * CDS: Parameter Value Component should display fail icon ✘
  * CDS: Parameter Value Component should display disabled icon ✘
  * CDS: Parameter Value Component should display skipped icon ✘
  * CDS: Parameter Value Component should display waiting icon ✘
  * CDS: Table component Table method ✘
  * CDS: Table component Null Table ✘


  * CDS: Variable Audit Component Load Component ✘
  * CDS: Variable From Component Create new variable ✘
  * CDS: Variable List Component Load Component &#43; update value ✘
  * CDS: Variable Value Component Load Component string ✘
  * CDS: Variable Value Component Load Component password ✘
  * CDS: Variable Value Component Load Component text ✘

* HeadlessChrome 0.0.0 (Linux 0.0.0).1245073
  * CDS: Pipeline Model should add new pipeline params ✘
  * CDS: Pipeline Model should keep old value ✘
  * CDS: Workflow Model should delete a node in the middle ✘
  * CDS: Workflow Model should delete root node: simple ✘
  * CDS: Workflow Model should not delete the root node because it has 2 triggers ✘
  * CDS: Workflow Model should delete the root node. O become the new root ✘
  * CDS: Workflow Model should delete root node, c because the new root ✘
  * CDS: Workflow Model should not delete the root node because it has 2 Joins ✘
  * CDS: Workflow Model should delete the node after join ✘
  * CDS: Workflow Model should delete the node after join ✘
  * CDS: Workflow Model should delete the node after join ✘

* HeadlessChrome 0.0.0 (Linux 0.0.0).1245074
  * CDS: application Store should create and delete an Application ✘





















* HeadlessChrome 0.0.0 (Linux 0.0.0).1245076
  * CDS: LoginComponent Click on Login button ✘
  * CDS: PasswordComponent Reset Password OK ✘
  * CDS: SignUPComponent SignUp OK ✘
  * CDS: VerifyComponent Verify OK ✘
  * CDS: Application Admin Component Load component &#43; renamed app ✘
  * CDS: Application Repo Component Load component &#43; select repository ✘
  * CDS: Application Load component &#43; load application ✘
  * CDS: Application Load component &#43; load application with error ✘
  * CDS: Application should run add variable ✘
  * CDS: Application should run update variable ✘
  * CDS: Application should run remove variable ✘
  * CDS: Application should run add permission ✘
  * CDS: Application should run update permission ✘
  * CDS: Application should run remove permission ✘
  * CDS: Application should run add/update/delete notification ✘
  * CDS: Pipeline Add Component should create an empty pipeline ✘
  * CDS: Pipeline Admin Component should update pipeline ✘
  * CDS: Pipeline Show should load component ✘
  * CDS: Pipeline Show should run add/update/delete permission ✘
  * CDS: Pipeline Show should run add/update/delete parameters ✘
  * CDS: Stage From component should add and delete prerequisite ✘









  * CDS: Artifact List should load component ✘
  * CDS: Test table component should load component &#43; filter testcase ✘
  * CDS: Test Report component should load component ✘

**Comment by bnjjj** [comment]
CDS Report root#4868.0

* Stage 1
  * run ✔

**Comment by bnjjj** [comment]
CDS Report ut-engine#4868.0

* Stage 1
  * Engine Test with DB ✔

**Comment by bnjjj** [comment]
CDS Report build-engine-cli-tools#4868.0

* Build minimal
  * cds cli legacy ✔ 
  * cdsctl linux ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔ 

* Build other os/arch
  * cds cli legacy darwin/windows amd64 -  
  * cdsctl darwin -  
  * cdsctl freebsd amd64 -  
  * Doc Build -  
  * Engine Build darwin amd64 -  
  * Engine Build freebsd amd64 -  
  * Engine Build linux arm -  
  * Engine Build windows amd64 -  
  * Plugins darwin amd64 -  
  * Plugins freebsd -  
  * Plugins linux amd64 -  
  * Plugins linux arm -  
  * Plugins windows amd64 -  
  * Worker darwin / windows / freebsd -  
  * Worker linux arm -  

* Package
  * Debian Engine Package ✔ 
  * Docker Engine Package ✔ 

* Publish
  * engine debian package ✔ 
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔

**Pull request merged**